includes/Jobs/JobRegistry.php

```diff
@@ -22,10 +26,12 @@ class JobRegistry {
 	 * Instantiate and init all jobs for the plugin.
 	 */
 	public function init() {
-		$action_scheduler_proxy = new ActionScheduler();
-
-		$this->generate_product_feed_job = new GenerateProductFeed( $action_scheduler_proxy );
+		$action_scheduler_proxy          = new ActionScheduler();
+		$feed_file_handler               = new FeedFileHandler();
+		$feed_data_exporter              = new FeedDataExporter();
```

Comment:
> or could they be abstracted away into GenerateProductFeed

That would be actually counter abstraction since now we are injecting dependencies to `GenerateProductFeed` which makes this a bit cleaner. What we could do is to have one additional Feed class that would assembly all of the things and would be used here instead. I don't really want to do that since it would be an abstraction for the sake of abstraction without a really clear benefit. JobRegistery is a nice place to assemble the elements required for the feed job.